### PR TITLE
`tjson`: Implement `datetime`

### LINKED ISSUE #905: `tjson`: Implement `datetime`
This task is a part of #683 epic.

🎯 The goal of the task is to implement `datetime` type for tjson.

Tigris' schema offers support for [date-time](https://docs.tigrisdata.com/overview/schema#date-time) presented according to RFC-3339. 

Substasks:
- [x] Support `datetime` storing it as Tigris' `date-time` format.

To Do:
- [x] Add unit tests
  - [x] Dedicated tests for datetime (like in the fjson package)
  - [x] Add datetime fields to one of the test cases in the `document_test.go` file.
- [ ] Port integration tests

### PR DESCRIPTION
**Pull request by noisersup**
# Description

This PR closes #905.

## Readiness checklist

* [x] I added tests for new functionality or bugfixes.
* [x] I ran `task all`, and it passed.
* [x] I added/updated comments for both exported and unexported top-level declarations (functions, types, etc).
* [ ] I checked comments rendering with `task godocs`.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

File: internal/tjson/datetime.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
+)
+
+// dateTimeType represents BSON datetime type.
+type dateTimeType time.Time
+
+// tjsontype implements tjsontype interface.
+func (dt *dateTimeType) tjsontype() {}
+
```
Comment (AlekSi) [inline internal/tjson/datetime.go:30]: Let's add `String() string` method just like in fjson

File: internal/tjson/datetime.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	var o time.Time
+	if err := dec.Decode(&o); err != nil {
+		return lazyerrors.Error(err)
+	}
+
+	if err := checkConsumed(dec, r); err != nil {
+		return lazyerrors.Error(err)
+	}
+
+	*dt = dateTimeType(o)
```
Comment (AlekSi) [inline internal/tjson/datetime.go:49]: Let's round it to milliseconds, that's what Tigris can store. Again, just like fjson

File: internal/tjson/datetime.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+	return nil
+}
+
+// MarshalJSON implements tjsontype interface.
+func (dt *dateTimeType) MarshalJSON() ([]byte, error) {
+	res, err := json.Marshal(time.Time(*dt))
```
Comment (AlekSi) [inline internal/tjson/datetime.go:56]: Round there too

**Review by AlekSi — COMMENTED** [review]
Looks good overall

**Review by AlekSi — APPROVED** [review]

**Review by w84thesun — APPROVED** [review]
LGTM

**Pull request merged**